Improve CSS support in server components (#1843)
Shopify/hydrogen: docs/framework/css-support.md

## How it works

> You can build with Tailwind's library or use other methods of styling like vanilla CSS or [CSS Modules](#css-modules). All the CSS imported in the app will be extracted at build time as a single asset.

Reviewer: ```suggestion
You can build with Tailwind's library or use other methods of styling like vanilla CSS or [CSS Modules](#css-modules). All CSS that's imported in the app is extracted as a single asset at build time.
```

3. Import your `.css` file into `index.html` or any desired client component.

## Using CSS

<aside class="note beta">
<h4>Experimental feature</h4>

<p>Importing CSS in React Server Components is an experimental feature. As a result, functionality is subject to change. You can provide feedback on this feature by <a href="https://github.com/Shopify/hydrogen/issues">submitting an issue in GitHub</a>.</p>

</aside>

> Hydrogen collects styles for each CSS file imported in your components, both in client and server components.

Reviewer: ```suggestion
Hydrogen collects styles for each CSS file imported in your client and server components.
```

> There are two modes for CSS support that can be modified by passing `experimental.css` option to the Hydrogen plugin in `vite.config.js`:

Reviewer: ```suggestion
You can modify the following modes for CSS support by passing `experimental.css` option to the Hydrogen plugin in `vite.config.js`:
```
.dev style preference is to avoid a count of the items being introduced. Maintenance issue.

{% codeblock file, filename: 'vite.config.js' %}

```
export default defineConfig({
  plugins: [hydrogen({experimental: {css: 'global'}})]
})
```

{% endcodeblock %}

> - `'modules-only'` mode: enables limited support for CSS Modules only. This is the current default mode.

Reviewer: ```suggestion
- **`modules-only`**: Enables limited support for CSS Modules only. This is the current default mode.
```

> - `'global'` mode: enables full support for both vanilla CSS and CSS Modules. It also enables a way to integrate with tools that provide CSS-in-JS at build time. This is the recommended mode.

Reviewer: ```suggestion
- **`global`**: Enables full support for vanilla CSS and CSS Modules. `global` also enables a way to integrate with tools that provide CSS-in-JS at build time. This is the recommended mode.
```

> Note that CSS code split for different routes is not supported at the moment.

Reviewer: ```suggestion
> Note:
> CSS code split for different routes is not currently supported.
```

### Vanilla (pure) CSS and extensions

> Vanilla CSS and language extensions such as [Sass](https://sass-lang.com/), [Less](https://lesscss.org/) or [Stylus](https://stylus-lang.com/) are supported under the `experimental.css: 'global'` feature.

Reviewer: ```suggestion
Vanilla CSS and language extensions such as [Sass](https://sass-lang.com/), [Less](https://lesscss.org/), and [Stylus](https://stylus-lang.com/) are supported under the `experimental.css: 'global'` feature.
```

> Once enabled, you can simply import your stylesheets directly in your server components:

Reviewer: ```suggestion
Once enabled, you can import your stylesheets directly in your server components:
```
.dev style prefers to avoid `simply`, I think because it makes an assumption that might not apply to all users.

{% endcodeblock %}

> When the CSS mode is `'modules-only'`, your styles will be inlined in a `<style>` tag before your component. This tag is only added automatically for the default export in the file. Consider using `'global'` CSS mode to support named exports and reduce code duplication.

Reviewer: ```suggestion
When the CSS mode is `modules-only`, styles are inlined in a `<style>` tag before your component. This tag is only added automatically for the default export in the file. Consider using the `global` CSS mode to support named exports and reduce code duplication.
```
Wonder if these paragraphs should be closer to where we define the modes, given that there's some detail here on when to use `global`.

Author: The idea is that we make `global` the default soon after making sure it works properly. Once we do that we can even remove any mention here about `experimental.css`.

### CSS-in-JS libraries

> CSS-in-JS libraries that emit `.css` files at **build time** can be supported by Hydrogen via third-party Vite plugins. Please, reach out to the library maintainers to ask for React Server Components support and feel free to tag the Hydrogen team.

Reviewer: ```suggestion
Hydrogen supports CSS-in-JS libraries that emit `.css` files at build time via third-party Vite plugins. Please, reach out to the library maintainers to ask for React Server Components support and feel free to tag the Hydrogen team.
```
.dev style reserved bold for UI elements

> On the other hand, CSS-in-JS libraries that collect styles at **runtime** are not supported at the moment due to limitations when integrating these libraries with React Server Components.

Reviewer: ```suggestion
However, CSS-in-JS libraries that collect styles at runtime aren't currently supported due to limitation integrating these libraries with React Server Components.
```
